apis/validator/block.v3.yaml

```diff
@@ -0,0 +1,92 @@
+get:
+  tags:
+    - ValidatorRequiredApi
+    - Validator
+  operationId: "produceBlockV3"
+  summary: "Produce a new block, without signature."
+  description: |
+    Requests a beacon node to produce a valid block, which can then be signed by a validator. The
+    returned block may be blinded or unblinded, depending on the current state of the network as
+    decided by the execution and beacon nodes.
+
+    The beacon node must return an unblinded block if it obtains the execution payload from its
+    paired execution node. It must only return a blinded block if it obtains the execution payload
+    header from an MEV relay.
+
+    Metadata in the response indicates the type of block produced, and the supported types of block
+    will be added to as forks progress.
+  parameters:
+    - name: slot
+      in: path
+      required: true
+      description: "The slot for which the block should be proposed."
+      schema:
+        $ref: "../../beacon-node-oapi.yaml#/components/schemas/Uint64"
+    - name: randao_reveal
+      in: query
+      required: true
+      description: "The validator's randao reveal value."
+      schema:
+        $ref: '../../beacon-node-oapi.yaml#/components/schemas/Signature'
+    - name: graffiti
+      in: query
+      required: false
+      description: "Arbitrary data validator wants to include in block."
+      schema:
+        $ref: '../../beacon-node-oapi.yaml#/components/schemas/Graffiti'
+    - name: skip_randao_verification
+      in: query
+      required: false
+      description: |
+        Skip verification of the `randao_reveal` value. If this flag is set then the
+        `randao_reveal` must be set to the point at infinity (`0xc0..00`). This query parameter
+        is a flag and does not take a value.
+      schema: {}
+      allowEmptyValue: true
+  responses:
+    "200":
+      description: Success response
+      headers:
+        Eth-Consensus-Version:
+          $ref: '../../beacon-node-oapi.yaml#/components/headers/Eth-Consensus-Version'
+        Eth-Blinded-Payload:
+          $ref: '../../beacon-node-oapi.yaml#/components/headers/Eth-Blinded-Payload'
+        Eth-Payload-Value:
+          $ref: '../../beacon-node-oapi.yaml#/components/headers/Eth-Payload-Value'
```

Comment:
Headers are required for SSZ returns. Should the added two headers be also present as JSON values to the `application/json` response type? Otherwise, why include version as property but not the payload value.

Reply:
Yeah, wasn't sure about that.  In the past we added the version to the JSON to allow for decoding, but it has pretty much been replaced with the header values now (as you don't need to do any mucking around with parsing streams to obtain the values from the headers).  I'm not averse to putting these new values in the JSON metadata, but not sure if they add significant value any more.

Reply:
> do we do something controversial and just return the json object not wrapped in a 'data' and have it just the json object or the ssz object determined by header?

It would definitely break the standard from other routes and prevent allowing to add more fields in the future. I would mirror the header values in the JSON type for overall consistency. It's easier to parse a JSON reply than headers for client side tooling that deals with JSON only

Reply:
@dapplion yeah I'm thinking the same.  If the metadata fields are in the JSON then it will allow parsing of the JSON separate from the HTTP response (e.g. if the JSON is saved and parsed later).  I'll update accordingly.

Reply:
Updated.  One question that rises from this is the naming of the fields: do we want to explicitly state which values are related to the execution payload, _i.e._

- `blinded-execution-payload` rather than `blinded-payload`
- `execution-payload-value` rather than `payload-value`

(and equivalent change for the header fields).  Obviously this makes them a little more verbose, but is safer if we ever end up with other payloads in the beacon block.  Thoughts?

Reply:
if we went that way, i'd probably suggest something like
```
execution-payload-value
execution-payload-blinded
```
i don't mind either way on whether we expand it...

Reply:
Have updated to use these names, agree that they are a bit clearer.  I think that this looks like the best way forward; 
@dapplion @rolfyone agree/disagree?